package/entrypoint.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+set -x
+
+if [ "$1" = "" ]; then
+    exec bin/${PROG} agent $@
+fi
+exec bin/${PROG} $@
```

Comment:
A couple of final nits, after testing this hands-on:
* I am a little sketched out by having the shell echo the exec command, as that will put things like datastore passwords in the log, if they're passing it as a CLI flag.
* The Dockerfile still has `CMD ["agent"]` in it to set the default command, so we don't need to handle that here, do we? 
* I think we need to quote `$@` and disable filename expansion to avoid re-splitting the args if they contain whitespace, otherwise we get:
```console
brandond@dev01:~/go/src/github.com/k3s-io/k3s$ docker run --rm -it --privileged rancher/k3s server "--etcd-snapshot-schedule-cron=0 */5 * * *"
exec bin/k3s server '--etcd-snapshot-schedule-cron=0' '*/5' bin dev entrypoint.sh etc lib proc run sys tmp var bin dev entrypoint.sh etc lib proc run sys tmp var bin dev entrypoint.sh etc lib proc run sys tmp var
```

```suggestion
#!/bin/sh -f
exec /bin/${PROG} "$@"
```